Assignment 3-Due 11.5.Rmd

```diff
@@ -0,0 +1,312 @@
+
+K4 <- data.frame(K3, fit1$cluster)
+
+#Have a look at the K4 dataframe. Lets change the names of the variables to make it more convenient with the names() command.
+
+names(K4) <- c("1", "2", "3", "4", "5", "cluster") 
+```
+
+Now we need to visualize the clusters we have created. To do so we want to play with the structure of our data. What would be most useful would be if we could visualize average motivation by cluster, by week. To do this we will need to convert our data from wide to long format. Remember your old friends tidyr and dplyr!
+
+First lets use tidyr to convert from wide to long format.
+```{r}
+
+K5 <- gather(K4, week, motivation, 1:5, factor_key=TRUE)
+K5
+```
+
+Now lets use dplyr to average our motivation values by week and by cluster.
+
+```{r}
+
+K6 <- K5%>%group_by( week,cluster)%>%summarise(mean=mean(motivation))%>%arrange(week, cluster)
+
+```
+
+Now it's time to do some visualization: 
+
+https://www.cs.uic.edu/~wilkinson/TheGrammarOfGraphics/GOG.html
+
+And you can see the range of available graphics in ggplot here:
+
+http://ggplot2.tidyverse.org/reference/index.html
+
+We are going to create a line plot similar to the one created in the school dropout paper we looked at in class (Bowers, 2010). It will have motivation on the Y-axis and weeks on the X-axis. To do this we will want our weeks variables to be treated as a number, but because it was created from a variable name it is currently being treated as a character variable. You can see this if you click on the arrow on the left of K6 in the Data pane. Week is designated by "chr". To convert it to numeric, we use the as.numeric command. 
+
+Likewise, since "cluster" is not numeric but rather a categorical label we want to convert it from an "integer" format to a "factor" format so that ggplot does not treat it as a number. We can do this with the as.factor() command.
+
+```{r}
+
+K6$week <- as.numeric(K6$week)
+
+K6$cluster <- as.factor(K6$cluster)
+
+```
+
+Now we can plot our line plot using the ggplot command, "ggplot()".
+
+- The first argument in a ggplot is the dataframe we are using: K6
+- Next is what is called an aesthetic (aes), the aesthetic tells ggplot which variables to use and how to use them. Here we are using the variables "week" and "avg" on the x and y axes and we are going color these variables using the "cluster" variable
+- Then we are going to tell ggplot which type of plot we want to use by specifiying a "geom()", in this case a line plot: geom_line()
+- Finally we are going to clean up our axes labels: xlab("Week") & ylab("Average Motivation")
+
+```{r}
+# install.packages("ggplot2")
+library(ggplot2)
+ggplot(K6, aes(week, mean, color = cluster)) +
+  geom_line() +
+  xlab("Week") + 
+  ylab("Average Motivation")
+
+```
+
+What patterns do you see in the plot?
+### Comments: Students with high motivation in the beginning became in low motivation and those who started withlow motivation became in high mood.
```

Comment:
What does this imply?

Reply:
Hi Anna,

Thank you for your comments. However, I am a little confused about how "deep" should we imply. Here I draw my interpretation from the trend of the plot, in which one line goes from high to low and the other goes from low to high. May I ask for more clues of how to "imply" more from the graph?

Thank you!

Best regards,
Jingru Zhang

Reply:
What I asked was what it means in the decision-making process if highly-motivated students lose their motivation while unmotivated students become motivated.

Reply:
Hi Anna,

Thanks a lot for the clue. It implies that the course cannot meet the expectation of the highly-motivated students because it is less challenging for them. The difficulty of the course should make a balance to meet different groups of students' capacity and expectation.

Best regards,
Jingru Zhang